lfs.c

```diff
@@ -2464,7 +2523,12 @@ int lfs_file_opencfg(lfs_t *lfs, lfs_file_t *file,
 
     // fetch attrs
     for (unsigned i = 0; i < file->cfg->attr_count; i++) {
-        if ((file->flags & 3) != LFS_O_WRONLY) {
+#ifdef LFS_READONLY
+        /* always LFS_O_RDONLY */
+#else
+        if ((file->flags & 3) != LFS_O_WRONLY)
+#endif /* LFS_READONLY */
+        {
```

Comment:
Hmm, this dangling squiggly bracket is a risk.

It looks like this can be changed to:
```
if ((file->flags & LFS_O_RDONLY) == LFS_O_RDONLY) {
```

Which would avoid the syntax issue.

Reply:
ok, reworked, but will now result in slightly bigger/slower code for read-only builds,
since `if ((file->flags & LFS_O_RDONLY) == LFS_O_RDONLY) {` will still be checked each time.

Reply:
fixed in 83fd2e8

Reply:
Hmm, that's a good point, but I don't know if the readability penalty is worth it